core/store/src/trie/iterator.rs

```diff
@@ -442,27 +442,138 @@ mod tests {
     use crate::trie::iterator::IterStep;
     use crate::trie::nibble_slice::NibbleSlice;
     use crate::Trie;
+    use near_o11y::testonly::init_test_logger;
     use near_primitives::shard_layout::ShardUId;
 
     fn value() -> Option<Vec<u8>> {
         Some(vec![0])
     }
 
+    fn verify_items_in_interval(
+        begin_key: &[u8],
+        end_key: &[u8],
+        trie: &Trie,
+        expected_keys: Vec<Vec<u8>>,
+    ) {
+        let path_begin: Vec<_> = NibbleSlice::new(begin_key).iter().collect();
+        let path_end: Vec<_> = NibbleSlice::new(end_key).iter().collect();
+        let mut trie_iter = trie.iter().unwrap();
+        let items = trie_iter.visit_nodes_interval(&path_begin, &path_end).unwrap();
+        let trie_item_keys: Vec<_> = items.into_iter().map(|item| item.key).flatten().collect();
+        let expected_item_keys: Vec<_> = expected_keys.into_iter().map(|item| item).collect();
+        assert_eq!(trie_item_keys, expected_item_keys);
+    }
+
     /// Checks that for visiting interval of trie nodes first state key is
     /// included and the last one is excluded.
     #[test]
-    fn test_visit_interval() {
+    fn test_visit_interval_basic() {
+        init_test_logger();
         let trie_changes = vec![(b"aa".to_vec(), Some(vec![1])), (b"abb".to_vec(), Some(vec![2]))];
         let tries = create_tries();
         let state_root =
             test_populate_trie(&tries, &Trie::EMPTY_ROOT, ShardUId::single_shard(), trie_changes);
         let trie = tries.get_trie_for_shard(ShardUId::single_shard(), state_root);
-        let path_begin: Vec<_> = NibbleSlice::new(b"aa").iter().collect();
-        let path_end: Vec<_> = NibbleSlice::new(b"abb").iter().collect();
-        let mut trie_iter = trie.iter().unwrap();
-        let items = trie_iter.visit_nodes_interval(&path_begin, &path_end).unwrap();
-        let trie_items: Vec<_> = items.into_iter().map(|item| item.key).flatten().collect();
-        assert_eq!(trie_items, vec![b"aa"]);
+
+        verify_items_in_interval(b"aa", b"abb", &trie, vec![b"aa".to_vec()]);
+    }
+
+    /// Checks that for visiting interval of trie nodes first state key is
+    /// included and the last one is excluded for degenerate trie tree
+    #[test]
+    fn test_visit_interval_degenerate() {
```

Comment:
nit: tests can be simplified even more, because they have common setup:

```
// generate_trie_changes
let tries = create_tries();
let state_root =
    test_populate_trie(&tries, &Trie::EMPTY_ROOT, ShardUId::single_shard(), trie_changes);
let trie = tries.get_trie_for_shard(ShardUId::single_shard(), state_root);
// verify_items_in_interval
```

Also, as we are not really using values, we can consider using `enumerate` over list of keys to auto-generate sequential values.